File: src/server/Model/Import/ImportMetadata.cs
// Copyright (c) 2021, UW Medicine Research IT, University of Washington
// Developed by Nic Dobbins and Cliff Spital, CRIO Sean Mooney
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.
using System;
using System.Collections.Generic;
using Model.Admin.Compiler;

namespace Model.Import
{
    public class ImportMetadata : IConstrainedResource
    {
        public Guid Id { get; set; }
        public string SourceId { get; set; }
        public ImportType Type { get; set; }
        public string StructureJson { get; set; }
        public IEnumerable<Constraint> Constraints { get; set; }
        public DateTime? Created { get; set; }
        public DateTime? Updated { get; set; }
    }

    public class ImportMetadataDTO : ImportMetadata
    {
        public ImportMetadata ToImportMetadata()
        {
            return new ImportMetadata
            {
                Id = Id,
                SourceId = SourceId,
                Type = Type,
                StructureJson = StructureJson,
                Constraints = Constraints,
                Created = Created,
                Updated = Updated
            };
        }
    }

    public enum ImportType
    {
        REDCapProject = 1,
        MRN = 2
    }
}

File: src/ui-client/src/models/dataImport/ImportMetadata.ts
/* Copyright (c) 2019, UW Medicine Research IT, University of Washington
 * Developed by Nic Dobbins and Cliff Spital, CRIO Sean Mooney
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { Concept } from "../concept/Concept";
import { BaseREDCapImportConfiguration } from "../redcapApi/ImportConfiguration";
import { Constraint } from "../admin/Concept";

export enum ImportType {
    REDCapProject = 1,
    MRN = 2
}

interface BaseImportMetadata {
    constraints: Constraint[];
    created: Date;
    id?: string;
    sourceId: string;
    type: ImportType;
    updated: Date;
}

export interface ImportMetadata extends BaseImportMetadata {
    structure: ImportStructure;
}

export interface ImportMetadataDTO extends BaseImportMetadata {
    structureJson: string;
}

export interface ImportStructure {
    id: string;
}

export interface REDCapImportStructure extends ImportStructure {
    concepts: Concept[];
    configuration: BaseREDCapImportConfiguration;
    patients: number;
}

export interface MrnImportStructure extends ImportStructure {
    name: string;
    category: string;
}